### PR TITLE
All configuration related code has been moved to a separate package

### LINKED ISSUE #270: cannot import source codes from external projects
The source code cannot be imported into external projects because they are at the root of the project.

It is recommended to move the source code to a pkg directory or to a separate package.

In particular I am interested in importing the configuration parser. It would be helpful to have such code openly.

Greetings

image

### PR DESCRIPTION
**Pull request by Eitol**
closes #270

### CODE REVIEW

**Comment by Eitol** [comment]
@hashicorp/api-tooling-core

**Comment by eikenb** [comment]
Hey @Eitol, sorry for the delay responding. Meant to review this today (I reserve Mondays to look at community messages) but didn't make it. I'll take a look tomorrow.

File: runner.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	config2 "github.com/hashicorp/envconsul/config"
```
Comment (eikenb) [inline runner.go:7]: Would you please consider aligning the naming of the imports so the module has the same name everywhere. You used `cfg` above which is fine and wouldn't conflict with the consul-template/config package. Please don't use `config2` (to close to existing `config` for my tastes).

File: runner_test.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package main
 
 import (
 	"fmt"
+	config2 "github.com/hashicorp/envconsul/config"
```
Comment (eikenb) [inline runner_test.go:5]: Same as above. Please use the same name everywhere.

**Pull request closed without merging**